### PR TITLE
Minor optimizations using private, params, and isEqualType

### PR DESCRIPTION
**Pull request by pennyworth12345**
**When merged this pull request will:**
- added the use of `params` in several files for setting variables from passed parameters
- replaced a few checks that were using `typeName` with `isEqualType` a
- removed all unneeded uses of `private "_var"`, and combined it with usage of `private` on the same line as the variable is declared
- removed and optimized some old and unused variables from `private` variable arrays

### CODE REVIEW

File: addons/atragmx/functions/fnc_target_speed_assist_timer.sqf [hunk 1]
```diff
@@ -23,9 +23,8 @@ if !(ctrlVisible 9000) then {
     ctrlSetFocus ((uiNamespace getVariable "ATragMX_Display") displayCtrl 9002);
 
     [{
-        private ["_args", "_startTime"];
-        _args = _this select 0;
-        _startTime = _args select 0;
+        private _args = _this select 0;
+        private _startTime = _args select 0;
```
Comment (jonpas) [inline addons/atragmx/functions/fnc_target_speed_assist_timer.sqf]: This should probably use `params` too:
```
params ["_args"];
_args params ["_startTime"];
```

File: addons/explosives/functions/fnc_addExplosiveActions.sqf [hunk 2]
```diff
@@ -36,8 +36,7 @@ _itemCount = [];
 _children = [];
 
 {
-    private "_name";
-    _name = if(isText(_x >> "displayNameShort") && {getText(_x >> "displayNameShort") != ""}) then
+    private _name = if(isText(_x >> "displayNameShort") && {getText(_x >> "displayNameShort") != ""}) then
         {getText (_x >> "displayNameShort")}else{getText(_x >> "displayName")};
```
Comment (jonpas) [inline addons/explosives/functions/fnc_addExplosiveActions.sqf]: This could be improved while at it.

File: addons/frag/functions/fnc_addTrack.sqf [hunk 3]
```diff
@@ -1,23 +1,20 @@
...
 if (GVAR(autoTrace)) then {
     [] call FUNC(startTracing);
 };
 
 // setAccTime 0.05;
-_index = (count GVAR(traces));
-_obj = _this select 1;
-_origin = _this select 0;
-_color = [1,0,0,1];
+private _index = (count GVAR(traces));
```
Comment (jonpas) [inline addons/frag/functions/fnc_addTrack.sqf]: No need for `()` here.

File: addons/frag/functions/fnc_doExplosions.sqf [hunk 4]
```diff
@@ -1,18 +1,12 @@
 //fnc_doExplosions.sqf
 #include "script_component.hpp"
 
-private ["_params", "_explosions", "_index", "_i", "_exp", "_refExp", "_bpos", "_hit", "_distance", "_indirectHitRange", "_depth"];
-_params = _this select 0;
-_explosions = _params select 0;
-_index = _params select 1;
+private _params = _this select 0;
```
Comment (jonpas) [inline addons/frag/functions/fnc_doExplosions.sqf]: `params ["_params"]`

File: addons/frag/functions/fnc_doSpall.sqf [hunk 5]
```diff
@@ -2,46 +2,42 @@
 #include "script_component.hpp"
 // ACE_player sideChat "WAAAAAAAAAAAAAAAAAAAAA";
 
-private ["_hitData", "_initialData", "_hpData", "_object", "_foundObjects", "_index", "_foundObjecsts", "_roundType", "_round", "_caliber", "_explosive", "_idh", "_alive", "_exit", "_vm", "_velocity", "_oldVelocity", "_curVelocity", "_diff", "_polar", "_unitDir", "_spallPos", "_pos1", "_i", "_pos2", "_blah", "_data", "_spallPolar", "_warn", "_c", "_m", "_k", "_gC", "_fragPower", "_fragTypes", "_spread", "_spallCount", "_elev", "_dir", "_vel", "_spallFragVect", "_fragType", "_fragment", "_pos"];
+private _hitData = _this select 0;
```
Comment (jonpas) [inline addons/frag/functions/fnc_doSpall.sqf]: `params ["_hitData"];`

File: addons/frag/functions/fnc_doSpall.sqf [hunk 6]
```diff
@@ -2,46 +2,42 @@
...
+private _initialData = GVAR(spallHPData) select (_hitData select 0);
+private _hpData = (_hitData select 1) select (_this select 1);
 
-_hitData = _this select 0;
-_initialData = GVAR(spallHPData) select (_hitData select 0);
-_hpData = (_hitData select 1) select (_this select 1);
 
-
-_object = _hpData select 0;
+private _object = _hpData select 0;
```
Comment (jonpas) [inline addons/frag/functions/fnc_doSpall.sqf]: `_hpData params ["_object"];`

File: addons/frag/functions/fnc_findReflections.sqf [hunk 7]
```diff
@@ -1,25 +1,18 @@
...
 BEGIN_COUNTER(fnc_findReflections);
-_params = _this select 0;
-_pos = _params select 0;
-_explosiveInfo = _params select 1;
-_los = _params select 2;
-_nlos = _params select 3;
-_zIndex = _params select 4;
-_depth = _params select 5;
-_rand = _params select 6;
+private _params = _this select 0;
```
Comment (jonpas) [inline addons/frag/functions/fnc_findReflections.sqf]: `params ["_params"];`

File: addons/frag/functions/fnc_spallTrack.sqf [hunk 8]
```diff
@@ -1,31 +1,27 @@
...
+private _intersectsWith = lineIntersectsWith [_curPos, _forwardPos];
 
 if (count _intersectsWith > 0) then {
     // player sideChat format["inter: %1", _intersectsWith];
     {
         if(!(_x in _foundObjects)) then {
             // diag_log text format["Adding HP: %1", _x];
-            _index = (count GVAR(spallHPData));
-            _hpId = _x addEventHandler ["hitPart", compile format["[%1, _this] call " + QFUNC(spallHP), _index]];
+            private _index = (count GVAR(spallHPData));
```
Comment (jonpas) [inline addons/frag/functions/fnc_spallTrack.sqf]: Excessive `()`.

File: addons/frag/functions/fnc_trackTrace.sqf [hunk 9]
```diff
@@ -1,13 +1,11 @@
 #include "script_component.hpp"
 
-private ["_params", "_tracerObj", "_index", "_positions", "_data"];
-_params = _this select 0;
-_tracerObj = _params select 0;
-_index = _params select 1;
+private _params = _this select 0;
```
Comment (jonpas) [inline addons/frag/functions/fnc_trackTrace.sqf]: `params ["_params"];`

File: addons/laser/functions/fnc_findStrongestRay.sqf [hunk 10]
```diff
@@ -1,13 +1,12 @@
...
-_spots = [];
-_outliers = [];
-_spot = [];
-_testPos = (_list select 0) select 0;
+
+params ["_list", "_checkPos"];
+private _spots = [];
+private _outliers = [];
+private _spot = [];
+private _testPos = (_list select 0) select 0;
```
Comment (jonpas) [inline addons/laser/functions/fnc_findStrongestRay.sqf]: `(_list select 0) params ["_testPos"];`

File: addons/laser/functions/fnc_findStrongestRay.sqf [hunk 11]
```diff
@@ -1,13 +1,12 @@
...
 {
-    _samplePos = _x select 0;
+    private _samplePos = _x select 0;
```
Comment (jonpas) [inline addons/laser/functions/fnc_findStrongestRay.sqf]: `_x params ["_samplePos"];`

File: addons/laser/functions/fnc_handleLaserOff.sqf [hunk 12]
```diff
@@ -1,8 +1,7 @@
 //fnc_handleLaserOff.sqf
 #include "script_component.hpp"
 
-private ["_uuid"];
-_uuid = _this select 0;
+private _uuid = _this select 0;
```
Comment (jonpas) [inline addons/laser/functions/fnc_handleLaserOff.sqf]: `params ["_uuid"];`

File: addons/medical/functions/fnc_treatmentIV.sqf [hunk 13]
```diff
@@ -16,12 +16,11 @@
 
 #include "script_component.hpp"
 
-private "_removeItem";
 params ["_caller", "_target", "_selectionName", "_className", "_items"];
 
 if (count _items == 0) exitWith {false};
 
-_removeItem = _items select 0;
+private _removeItem = _items select 0;
```
Comment (jonpas) [inline addons/medical/functions/fnc_treatmentIV.sqf]: `(_items select 0) params ["_removeItem"];`

File: addons/parachute/functions/fnc_checkCutParachute.sqf [hunk 14]
```diff
@@ -14,6 +14,5 @@
  * Public: No
  */
 #include "script_component.hpp"
-private["_unit"];
-_unit = _this select 0;
+private _unit = _this select 0;
```
Comment (jonpas) [inline addons/parachute/functions/fnc_checkCutParachute.sqf]: `params ["_unit"];`

File: addons/parachute/functions/fnc_cutParachute.sqf [hunk 15]
```diff
@@ -14,9 +14,8 @@
  * Public: No
  */
 #include "script_component.hpp"
-private["_unit","_vehicle"];
-_unit = _this select 0;
-_vehicle = vehicle _unit;
+private _unit = _this select 0;
```
Comment (jonpas) [inline addons/parachute/functions/fnc_cutParachute.sqf]: `params ["_unit"];`

File: addons/rearm/XEH_respawn.sqf [hunk 16]
```diff
@@ -1,14 +1,12 @@
 #include "script_component.hpp"
 
-private ["_unit"];
-
-_unit = _this select 0;
+private _unit = _this select 0;
```
Comment (jonpas) [inline addons/rearm/XEH_respawn.sqf]: `params ["_unit"];`

File: addons/refuel/XEH_respawn.sqf [hunk 17]
```diff
@@ -1,8 +1,6 @@
 #include "script_component.hpp"
 
-private ["_unit"];
-
-_unit = _this select 0;
+private _unit = _this select 0;
```
Comment (jonpas) [inline addons/refuel/XEH_respawn.sqf]: `params ["_unit"];`

File: addons/vector/functions/fnc_convertToTexturesDegree.sqf [hunk 18]
```diff
@@ -6,9 +6,7 @@ converts azimuth into array of textures for the vectors scripted info control
 
 #include "script_component.hpp"
 
-private "_number";
-
-_number = _this select 0;
+private _number = _this select 0;
```
Comment (jonpas) [inline addons/vector/functions/fnc_convertToTexturesDegree.sqf]: `params ["_number"];`

**Comment by pennyworth12345** [comment]
From my testing, using `private` and `select` are marginally faster than `params` when used with one argument. I do agree though that `params` should be used as it's cleaner and more uniform with the rest of ACE.

File: addons/medical/functions/fnc_treatmentIV.sqf (hunk 13 above)
Comment (pennyworth12345) [inline addons/medical/functions/fnc_treatmentIV.sqf]: Shouldn't it be this:
```
_items params ["_removeItem"];
```

File: addons/explosives/functions/fnc_addExplosiveActions.sqf (hunk 2 above)
Comment (commy2) [inline addons/explosives/functions/fnc_addExplosiveActions.sqf]: I would've written this as:

```sqf
    private _name = getText (_x >> "displayNameShort");

    if (_name isEqualTo "") then {
        _name = getText (_x >> "displayName");
    };

    _children pushBack [
```

displayNameShort is always a string or undefined. the game would complain otherwise.

File: addons/atragmx/functions/fnc_calculate_range_card.sqf [hunk 19]
```diff
@@ -17,64 +17,58 @@
...
-_atmosphereModel = GVAR(workingMemory) select 17;
+private _bulletMass = GVAR(workingMemory) select 12;
+private _bulletDiameter = GVAR(workingMemory) select 13;
+private _boreHeight = GVAR(workingMemory) select 5;
+private _airFriction = GVAR(workingMemory) select 4;
+private _barrelTwist = GVAR(workingMemory) select 14;
+private _muzzleVelocity = GVAR(workingMemory) select 1;
+private _bc = GVAR(workingMemory) select 15;
+private _dragModel = GVAR(workingMemory) select 16;
+private _atmosphereModel = GVAR(workingMemory) select 17;
```
Comment (commy2) [inline addons/atragmx/functions/fnc_calculate_range_card.sqf]:
```sqf
GVAR(workingMemory) params [
    "",
    "_muzzleVelocity",
    "",
    "_scopeBaseAngle",
    "_airFriction",
    "_boreHeight",
    "",
    "",
    "",
    "",
    "",
    "",
    "_bulletMass",
    "_bulletDiameter",
    "_barrelTwist",
    "_bc",
    "_dragModel",
    "_atmosphereModel"
];
```

File: addons/vehiclelock/functions/fnc_setVehicleLockEH.sqf [hunk 20]
```diff
@@ -16,11 +16,9 @@
  */
 #include "script_component.hpp"
 
-private ["_lockNumber"];
-
 params ["_veh", "_isLocked"];
 TRACE_2("params",_veh,_isLocked);
 
-_lockNumber = if (_isLocked) then {2} else {0};
+private _lockNumber = if (_isLocked) then {2} else {0};
```
Comment (commy2) [inline addons/vehiclelock/functions/fnc_setVehicleLockEH.sqf]: Might as well change this to:
```sqf
private _lockNumber = [0, 2] select _isLocked;
```

File: addons/explosives/functions/fnc_setSpeedDial.sqf [hunk 21]
```diff
@@ -15,11 +15,10 @@
...
  */
 #include "script_component.hpp"
-private ["_speedDial", "_amount"];
 
-_speedDial = ace_player getVariable [QGVAR(SpeedDial), []];
-if (count _speedDial == 0) exitWith {};
-_amount = if((_this select 0))then{1}else{-1};
+private _speedDial = ace_player getVariable [QGVAR(SpeedDial), []];
+if (_speedDial isEqualTo []) exitWith {};
+private _amount = if((_this select 0))then{1}else{-1};
```
Comment (commy2) [inline addons/explosives/functions/fnc_setSpeedDial.sqf]:
```sqf
private _amount = [-1, 1] select (_this select 0);
```

File: addons/explosives/functions/fnc_setSpeedDial.sqf [hunk 22]
```diff
@@ -15,11 +15,10 @@
  * Public: No
  */
 #include "script_component.hpp"
-private ["_speedDial", "_amount"];
 
-_speedDial = ace_player getVariable [QGVAR(SpeedDial), []];
-if (count _speedDial == 0) exitWith {};
-_amount = if((_this select 0))then{1}else{-1};
+private _speedDial = ace_player getVariable [QGVAR(SpeedDial), []];
```
Comment (commy2) [inline addons/explosives/functions/fnc_setSpeedDial.sqf]: `ACE_player`

File: addons/concertina_wire/functions/fnc_dismount.sqf [hunk 23]
```diff
@@ -22,9 +22,8 @@ if (uiNamespace getVariable [QEGVAR(interact_menu,cursorMenuOpened),false]) exit
 };
 params ["_wire", "_unit"];
 
-private ["_config", "_delay"];
-_config = (configFile >> "CfgVehicles" >> typeOf _unit);
-_delay = if (getNumber(_config >> "engineer") == 1 || getNumber(_config >> "canDeactivateMines") == 1) then {60} else {120};
+private _config = (configFile >> "CfgVehicles" >> typeOf _unit);
+private _delay = if (getNumber(_config >> "engineer") == 1 || getNumber(_config >> "canDeactivateMines") == 1) then {60} else {120};
```
Comment (commy2) [inline addons/concertina_wire/functions/fnc_dismount.sqf]:
```sqf
private _delay = [120, 60] select (getNumber (_config >> "engineer") == 1 || {getNumber (_config >> "canDeactivateMines") == 1});
```

File: addons/concertina_wire/functions/fnc_dismount.sqf [hunk 24]
```diff
@@ -22,9 +22,8 @@ if (uiNamespace getVariable [QEGVAR(interact_menu,cursorMenuOpened),false]) exit
 };
 params ["_wire", "_unit"];
 
-private ["_config", "_delay"];
-_config = (configFile >> "CfgVehicles" >> typeOf _unit);
-_delay = if (getNumber(_config >> "engineer") == 1 || getNumber(_config >> "canDeactivateMines") == 1) then {60} else {120};
+private _config = (configFile >> "CfgVehicles" >> typeOf _unit);
```
Comment (commy2) [inline addons/concertina_wire/functions/fnc_dismount.sqf]: superfluous parenthesis

File: addons/concertina_wire/functions/fnc_deploy.sqf [hunk 25]
```diff
@@ -39,16 +37,15 @@ GVAR(deployPFH) = [{
...
+    private _posStart = (_wireNoGeo modelToWorldVisual (_wireNoGeo selectionPosition "start")) call EFUNC(common,positionToASL);
+    private _posEnd = (getPosASL _unit) vectorAdd (vectorDir _unit);
+    private _dirVect = _posStart vectorDiff _posEnd;
+    private _dir = _dirVect call CBA_fnc_vectDir;
+    private _range = vectorMagnitude _dirVect;
+    private _anim = 0 max (1 - (_range / 12));
 
     if (!(alive _unit) || _range >= 12 || (_unit getVariable [QGVAR(wireDeployed), false])) exitWith {
-        _wire = "ACE_ConcertinaWire" createvehicle [0, 0, 0];
+        private _wire = "ACE_ConcertinaWire" createvehicle [0, 0, 0];
```
Comment (commy2) [inline addons/concertina_wire/functions/fnc_deploy.sqf]: `createVehicle`

**Comment by Glowbal** [comment]
This PR is slightly larger as I like. @acemod/devs could you please help double check this on scope changes? It's something that has come back to haunt us last year on similar clean up PRs.

Because of the size of this PR and no added functionality, I'd like to fast track this.

Thanks for making this @pennyworth12345.

File: addons/concertina_wire/functions/fnc_dismount.sqf (hunk 24 above)
Comment (Glowbal) [inline addons/concertina_wire/functions/fnc_dismount.sqf]: Not part of this PR

File: addons/vector/functions/fnc_convertToTexturesDegree.sqf [hunk 26]
```diff
@@ -6,38 +6,35 @@ converts azimuth into array of textures for the vectors scripted info control
...
-private "_number";
-
-_number = _this select 0;
+prams ["_number"];
 
 if (GVAR(useMil)) then {
 
     _number = round (6400 / 360 * _number);
-    if (_number == 6400) then {_number = 0};
+    if (_number isEqualTo 6400) then {_number = 0};
```
Comment (VKing6) [inline addons/vector/functions/fnc_convertToTexturesDegree.sqf]: Replacing == with isEqualTo wasn't mentioned in the PR.
Intended/Ok?

Comment (jonpas) [inline addons/vector/functions/fnc_convertToTexturesDegree.sqf]: Is OK, was added after initial PR was written.

**Comment by Glowbal** [comment]
Replaced by #4323.  Contains all original work + resolved merge conflicts.
